### PR TITLE
Bug 1801095: Fix imagepolicyresolve plugin to resolve when enabled on an existing object

### PR DESCRIPTION
**Pull request by tnozicka**
/cc @soltysh @adambkaplan @smarterclayton 

/hold
for proof in https://github.com/openshift/origin/pull/24530

### CODE REVIEW

File: pkg/admission/imagepolicy/imagereferencemutators/meta.go [hunk 1]
```diff
@@ -7,16 +7,44 @@ import (
...
 
 var errNoImageMutator = fmt.Errorf("No list of images available for this object")
 
+func ResolveAllNames(annotations AnnotationAccessor) bool {
+	if annotations == nil {
+		return false
+	}
+
+	a, ok := annotations.TemplateAnnotations()
+	if ok {
```
Comment (soltysh) [inline pkg/admission/imagepolicy/imagereferencemutators/meta.go]: Squash these 2 ifs together into:
```
if ok && a[imagepolicyv1.ResolveNamesAnnotation] == "*" {
    return true
}
```

File: pkg/admission/imagepolicy/imagereferencemutators/meta.go [hunk 2]
```diff
@@ -7,16 +7,44 @@ import (
...
+		if a[imagepolicyv1.ResolveNamesAnnotation] == "*" {
+			return true
+		}
+	}
+
+	a = annotations.Annotations()
+	if a != nil {
```
Comment (soltysh) [inline pkg/admission/imagepolicy/imagereferencemutators/meta.go]: Ditto.

File: pkg/admission/imagepolicy/imagereferencemutators/meta.go [hunk 3]
```diff
@@ -7,16 +7,44 @@ import (
...
+}
+
 // GetImageReferenceMutator returns a mutator for the provided object, or an error if no
 // such mutator is defined. Only references that are different between obj and old will
 // be returned unless old is nil.
-func (KubeImageMutators) GetImageReferenceMutator(obj, old runtime.Object) (ImageReferenceMutator, error) {
+func (m KubeImageMutators) GetImageReferenceMutator(obj, old runtime.Object) (ImageReferenceMutator, error) {
+	oldAnnotations, _ := m.GetAnnotationAccessor(old)
+	annotations, _ := m.GetAnnotationAccessor(obj)
+	edge := ResolveAllNames(annotations) != ResolveAllNames(oldAnnotations)
```
Comment (soltysh) [inline pkg/admission/imagepolicy/imagereferencemutators/meta.go]: Maybe `update` or better fitting name?

**Comment by tnozicka** [comment]
green tests in https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24530/pull-ci-openshift-origin-master-e2e-aws-image-registry/1080

/hold cancel

**Pull request merged**